### PR TITLE
helix: update to 24.07

### PR DESCRIPTION
**Pull request by avnyu**
icon.patch: use windres to add icon when target_env is gnu
disable-lto.patch: PKGBUILD failed at cargo test --locked --release, alternatively test without --release can work too
This is my first time to create a pull request, sorry in advanced if I cause any problems.

### CODE REVIEW

File: mingw-w64-helix/PKGBUILD [hunk 1]
```diff
@@ -12,11 +12,16 @@ url="https://github.com/helix-editor/helix"
 license=('spdx:MPL-2.0')
 makedepends=("${MINGW_PACKAGE_PREFIX}-rust"
              "${MINGW_PACKAGE_PREFIX}-mdbook"
+             "patch"
```
Comment (ognevny) [inline mingw-w64-helix/PKGBUILD:15]: there is no need to add this package. it's always installed in build machines

**Comment by ognevny** [comment]
can be icon.patched added upstream?

File: mingw-w64-helix/icon.patch [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+         println!("cargo:rustc-link-lib=dylib=resource");
+     }
+ 
++    #[cfg(target_env = "gnu")]
++    fn compile_with_windres_gnu(output: PathBuf, input: PathBuf) {
++        let mut command = process::Command::new("windres.exe");
++        let command = command.arg(format!(
++            "--include-dir={}",
++            env::var("CARGO_MANIFEST_DIR")
++                .expect("CARGO_MANIFEST_DIR should have been set by Cargo")
```
Comment (ognevny) [inline mingw-w64-helix/icon.patch:34]:
```suggestion
+                .unwrap()
```
nit: `Result::Err` won't be returned in this case, no need to handle it woth `expect`

**Comment by ognevny** [comment]
don't forget to run `updpkgsums` after changing files :)

**Comment by avnyu** [comment]
sorry about that, about the icon.patch, it should work upstream, but i didn't tested

**Comment by ognevny** [comment]
add `${MINGW_PACKAGE_PREFIX}-gcc-libs` to depends

**Comment by ognevny** [comment]
I tested it on CLANG64. sorry for delay, thanks for contribution

**Pull request merged**